src/main.cpp

```diff
@@ -2181,8 +2181,8 @@ bool ConnectBlock(const CBlock &block,
     {
         if (pindex->GetUndoPos().IsNull())
         {
-            CDiskBlockPos pos;
-            if (!FindUndoPos(state, pindex->nFile, pos, ::GetSerializeSize(blockundo, SER_DISK, CLIENT_VERSION) + 40))
+            CDiskBlockPos pos2;
+            if (!FindUndoPos(state, pindex->nFile, pos2, ::GetSerializeSize(blockundo, SER_DISK, CLIENT_VERSION) + 40))
```

Comment:
possibly an nit, but using pos2 seems to say there there's another one around nearby but really the it's in FindUndoPos() and FindBlockPos()  ... that's where the shadowing is happening.  It think it would be clearer to use _pos in those functions. 

-bool FindUndoPos(CValidationState &state, int nFile, CDiskBlockPos &pos, unsigned int nAddSize)
+bool FindUndoPos(CValidationState &state, int nFile, CDiskBlockPos &_pos, unsigned int nAddSize)

Reply:
According to `clang` analyzer the shadowin happen here:

Reply:
According to `clang` analyzer the shadowing happen here: 

```
    main.cpp:2183:27: warning: declaration shadows a local variable [-Wshadow]
                CDiskBlockPos pos;
                              ^
    main.cpp:1963:16: note: previous declaration is here
        CDiskTxPos pos(pindex->GetBlockPos(), GetSizeOfCompactSize(block.vtx.size()));
                   ^
```